testutil/compose/alert.go

```diff
@@ -40,8 +40,11 @@ func startAlertCollector(ctx context.Context, dir string) chan string {
 			dedup   = make(map[string]bool)
 		)
 		defer close(resp)
+
+		// Time required to wait for prometheus container to start.
+		time.Sleep(time.Second * 10)
 		for ctx.Err() == nil {
-			time.Sleep(time.Second * 5)
+			time.Sleep(time.Second * 2)
```

Comment:
rather sleep at the end of the loop (so you check for context closed directly after sleeping instead of doing logic with a possible cancelled context)